src/aerie_cli/commands/command_context.py

```diff
@@ -1,17 +1,37 @@
 import typer
 from aerie_cli.aerie_client import AerieClient
 from aerie_cli.utils.sessions import get_active_session_client
+from aerie_cli.aerie_host import AerieHostConfiguration
 
 app = typer.Typer()
 
 class CommandContext:
-    hasura_admin_secret = None
+    hasura_admin_secret: str = None
+    configuration: AerieHostConfiguration = None
+    __client: AerieClient = None
+
     def __init__(self) -> None:
         raise NotImplementedError
 
     @classmethod
     def get_client(cls) -> AerieClient:
-        client = get_active_session_client()
+        """Get the AerieClient for any command's execution.
+        
+        Returns:
+            AerieClient
+        """
+        # If the configuration was set in the CLI by the user,
+        # then the returned client will be derived from that configuration.
+        if cls.configuration != None and cls.__client == None:
+            session = cls.configuration.start_session()
```

Comment:
You'll need to add auth credentials here if the host requires authentication. Suggest you maybe refactor these lines into a separate method, maybe in `utils/sessions.py`:

https://github.com/NASA-AMMOS/aerie-cli/blob/5c894779d796ab6a244d7c22bd4b541b966602a8/src/aerie_cli/commands/configurations.py#L153-L163

Reply:
I refactored the code you mentioned into the new `start_session_from_configuration` in `utils.sessions.py`. Username/password can now be entered with the alternate configuration options if needed.

These changes are in 58967ad